src/irr/core/LRUcache.h

```diff
@@ -0,0 +1,58 @@
+#include <list>
+#include "irr/core/Types.h"
+#include "COpenGLDriver.h"
+
+namespace irr {
+	namespace core {
+
+		template<typename Key, typename Value,typename MapHash,typename MapEquals>
+		class LRUcache {
+			list<Key> stored_keys;
+			unordered_map<Key, Value, MapHash,MapEquals> map;
+			uint32_t cap;
+			std::atomic_uint32_t timestamp;
+		public:
+			inline LRUcache() = default;
+			inline LRUcache(uint32_t& capacity)
+			{
+				cap = capacity;
+			}
+
+			inline void insert(Key &k, Value &v)
+			{
+				if (map.find(k) == map.end() && stored_keys.size() == cap) {
+					auto last = stored_keys.back();
+					stored_keys.pop_back();
+					map.erase(last);
+				}
+				else
+					stored_keys.remove(k);
+				stored_keys.push_front(k);
```

Comment:
this works wrong.

If the cap would be overflown, we always need to remove the oldest value.

See here
https://www.geeksforgeeks.org/lru-cache-implementation/